src/mscorlib/src/System/Environment.cs

```diff
@@ -1338,6 +1394,8 @@ internal static String GetStackTrace(Exception e, bool needFileInfo)
 #if BIT64
                     // 64-bit programs run only on 64-bit
                     return true;
+#elif PLATFORM_UNIX
+                    return IntPtr.Size == 8; // TODO: use uname and match utsname.machine?
```

Comment:
We have the same simplifying assumption that OS architecture == Process architecture in `System.Runtime.InteropServices.RuntimeInformation`: 

https://github.com/dotnet/corefx/blob/d0dc5fc099946adc1035b34a8b1f6042eddb0c75/src/Native/System.Native/pal_runtimeinformation.cpp#L37

The same simplifying assumption should apply here.